### DISCUSSION TITLE
Sub navigation

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#714] [RELEASED: Components, pages and patterns]
The [sub navigation component](https://design-patterns.service.justice.gov.uk/components/sub-navigation/) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by helennickols** [comment]
[Slack discussion: sub-nav examples (MoJ only)](https://mojdt.slack.com/archives/CH5RUSB27/p1740567500316999)

**Comment by rwk-moj** [comment]
Hi all,

Just adding some feedback here that we received around the Sub Navigation from an accessibility audit. 

It has been put to us that the contrast for the focus state does not meet sufficient contrast ratios.  This is when the tabbed item takes focus and displays a yellow focus on a white background (see screenshot below).

> The focus indicator throughout the site fails to meet minimum contrast requirements with a contrast ratio of 1.3:1 (with a foreground yellow colour of #FFDD00 and a background white colour of #FFFFFF).

> Elements receiving focus should have a minimum 3:1 contrast ratio against the background. Low contrast controls are more difficult to perceive, and may be completely missed by people with a visual impairment. Insufficient contrast can be problematic in other situations as well, such as when users are using their device in sunlight, or with the brightness turned down.

> While there are some elements where the yellow focus indicator is accompanied by a thick black underline or outline, there are other elements (such as the pictured tab control) that do not implement this. Therefore, for some keyboard users with low vision, there would be no visual difference between the focused and non-focused state.

> This is also related to WCAG 2.2 (to be released) success criterion 2.4.11 - Focus Appearance, which states that the focus indicator has a contrast ratio of at least 3:1 against adjacent colours.

![image](https://github.com/user-attachments/assets/3babebc0-cf6e-4564-9b6c-ba59e139e19a)

We have reviewed the GDS Tabs component and the <a> tag takes the focus instead of the whole tab area. 

Ouur team could put forward a solution or fix for this but we'd need build this into our roadmap and priorities.. We also felt that it would be better placed to come from the design system / pattern library team to investigate and decide whether it needs to be fixed. 

## MoJ Pattern Library - Sub navigation focus state
![image](https://github.com/user-attachments/assets/6bdc1bc1-2641-434c-a5a8-9414fff32ddc)

## GDS Design System - Tabs focus state 
![image](https://github.com/user-attachments/assets/c1b8d7bb-d866-4d11-abe6-b7b1cb516bc4)

**Comment by Whersi** [comment]
##  Suggest a change form - Andreea Vlad

> It will be useful to have the full-functionality for the 'Sub-navigation' available. The code includes just the top area and I would find it useful to extend to the functionality where by clicking on a link, the info to change from one link to another. This will be consistent with gov.uk design system; for example, https://design-system.service.gov.uk/components/tabs/

**Comment by jamiehylands** [comment]
Defra Livestock Information Service

Hi, 

We had a requirement for a horizontal navigation component which differed from the GDS tabs UI. We selected the MOJ sub-navigation as the foundation before ingesting it into our Defra component library. 

We have now completed several rounds of user research and feedback has been universally positive. We have a large cohort often with low scores in digital inclusion so to find a component which helped them to complete their goals was excellent. 

<img width="1030" height="655" alt="Screenshot 2026-09-16 at 12 10 04" src="https://github.com/user-attachments/assets/b235fc3b-882d-4f2b-9034-adc0ce1a7090" />
